mlflow/models/signature.py

```diff
@@ -4,22 +4,24 @@
 Model signature defines schema of model input and output. See :py:class:`mlflow.types.schema.Schema`
 for more details on Schema and data types.
 """
-from typing import Dict, Any, Union
+from typing import Dict, Any, Union, TYPE_CHECKING
 
 import pandas as pd
 import numpy as np
 
 from mlflow.types.schema import Schema
 from mlflow.types.utils import _infer_schema
 
-try:
-    import pyspark.sql.dataframe
+# At runtime, we don't need  `pyspark.sql.dataframe`
+if TYPE_CHECKING:
```

Comment:
Could you explain when "TYPE_CHECKING" is True ? When running type check ? When doing type checking will it import pyspark ?

Reply:
I roughly search for this, seems when running type checking no code will be executed. 
> The type checker never executes your code: instead, it analyzes it. Type checkers are implemented in pretty much the same way compilers are implemented, minus the "generate bytecode/assembly/machine code" step.

But if it don't execute code, how does it enter the `if` branch ?

Reply:
When `mypy` analyzes code, it does recognize modules under `if TYPE_CHECKING:`.

```python
from typing import TYPE_CHECKING


if TYPE_CHECKING:
    import foo
```

```
% mypy a.py
a.py:5: error: Cannot find implementation or library stub for module named 'foo'
a.py:5: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1 source file)
```